app/jobs/prepare_move_notifications_job.rb

```diff
@@ -4,22 +4,25 @@
 class PrepareMoveNotificationsJob < ApplicationJob
   queue_as :notifications
 
-  def perform(topic_id:, action_name:)
+  def perform(topic_id:, action_name:, send_webhooks: true, send_emails: true, only_supplier_id: nil)
     move = Move.find(topic_id)
 
-    move.suppliers.each do |supplier|
+    # if the move has a specified supplier, use it; otherwise use the move.suppliers delegate based on from_location
+    [move.supplier || move.suppliers].flatten.each do |supplier|
```

Comment:
is there any chance we'll ever have one of each at the same time? Maybe we should merge if that's the case, otherwise this is fine

Reply:
I think the logic should be: use the `move.supplier_id` if it exists, otherwise infer the supplier from the move's `to_location`: that is how this is currently written. I don't think we should be using _both_ the supplier_id and the inferred suppliers